save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/ContestController.kt

```diff
@@ -54,4 +70,24 @@ internal class ContestController(
     ): Flux<ContestDto> = Flux.fromIterable(
         contestService.findFinishedContests(pageSize)
     ).map { it.toDto() }
+
+    /**
+     * @param contestName
+     * @return Organization
+     */
+    @GetMapping("/{contestName}/public-test")
+    @PreAuthorize("permitAll()")
+    @Suppress("UnsafeCallOnNullableType")
+    fun getPublicTestForContest(
+        @PathVariable contestName: String,
+    ): Mono<PublicTestDto> {
+        val contest = contestService.findByName(contestName).get()
+        val testSuite = testSuitesService.findTestSuiteById(
+            contest.testSuiteIds?.split(",")?.first()?.toLong()!!
+        ).get()
+        val test = testService.findTestsByTestSuiteId(testSuite.id!!).first()
+        val testRequest = TestFilesRequest(test.toDto(), testSuite.testRootPath)
+        return preprocessorWebClient.post().uri("/getTest").bodyValue(testRequest)
+            .retrieve().bodyToMono(PublicTestDto::class.java)
```

Comment:
There is an extension method that accepts a reified type parameter instead of a class